From 7a14f70c57d7c2d90512d8f629becdac4a0a540d Mon Sep 17 00:00:00 2001
From: Rahul T R <r-ravikumar@ti.com>
Date: Tue, 26 Mar 2024 15:48:27 +0530
Subject: [PATCH] arm64: dts: ti: Add DTB overlay for vision apps

Confine the memory map changes into a separate rtos-memory-map.dts
which describes the common memory map for all RTOS remoteprocs.

vision-apps.dtbo uses this and also disables capture and display
related peripherals from Linux so that it can be used by RTOS.

Also, Update the linux CMA region to allocate the required size in the
same region as the tiovx shared memory region. This is required to allow
the codec to access tiovx buffers through DMABUF IMPORT feature.

Signed-off-by: Aniket Limaye <a-limaye@ti.com>
Signed-off-by: Rahul T R <r-ravikumar@ti.com>
---
 arch/arm64/boot/dts/ti/Makefile               |   1 +
 .../dts/ti/k3-j721s2-rtos-memory-map.dtsi     | 244 ++++++++++++++++++
 .../boot/dts/ti/k3-j721s2-vision-apps.dtso    |  49 ++++
 3 files changed, 294 insertions(+)
 create mode 100644 arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi
 create mode 100644 arch/arm64/boot/dts/ti/k3-j721s2-vision-apps.dtso

diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
index 18ec74a853c5..157dee212b61 100644
--- a/arch/arm64/boot/dts/ti/Makefile
+++ b/arch/arm64/boot/dts/ti/Makefile
@@ -137,6 +137,7 @@ dtb-$(CONFIG_ARCH_K3) += k3-j721s2-evm-gesi-exp-board.dtbo
 k3-j721s2-evm-dtbs := k3-j721s2-common-proc-board.dtb k3-j721s2-evm-gesi-exp-board.dtbo
 dtb-$(CONFIG_ARCH_K3) += k3-j721s2-evm.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-j721s2-evm-pcie1-ep.dtbo
+dtb-$(CONFIG_ARCH_K3) += k3-j721s2-vision-apps.dtbo
 
 # Boards with J722s SoC
 dtb-$(CONFIG_ARCH_K3) += k3-j722s-evm.dtb
diff --git a/arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi b/arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi
new file mode 100644
index 000000000000..702055e6c2f4
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Vision-apps: device-tree overlay
+ *
+ * Copyright (C) 2019 Texas Instruments Incorporated - http://www.ti.com/
+ */
+
+#include <dt-bindings/soc/ti,sci_pm_domain.h>
+
+/dts-v1/;
+/plugin/;
+
+&{/} {
+	dma_buf_phys {
+		compatible = "ti,dma-buf-phys";
+	};
+};
+
+&mcu_r5fss0_core0_memory_region {
+	status = "disabled";
+};
+
+&mcu_r5fss0_core0_dma_memory_region {
+	status = "disabled";
+};
+
+&mcu_r5fss0_core1_dma_memory_region {
+	status = "disabled";
+};
+
+&mcu_r5fss0_core1_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core0_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core0_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core1_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss0_core1_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core0_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core0_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core1_dma_memory_region {
+	status = "disabled";
+};
+
+&main_r5fss1_core1_memory_region {
+	status = "disabled";
+};
+
+&c71_0_dma_memory_region {
+	status = "disabled";
+};
+
+&c71_0_memory_region {
+	status = "disabled";
+};
+
+&c71_1_dma_memory_region {
+	status = "disabled";
+};
+
+&c71_1_memory_region {
+	status = "disabled";
+};
+
+&rtos_ipc_memory_region {
+	status = "disabled";
+};
+
+&reserved_memory {
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	vision_apps_mcu_r5fss0_core0_dma_memory_region: vision-apps-r5f-dma-memory@a0000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa0000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_mcu_r5fss0_core0_memory_region: vision-apps-r5f-memory@a0100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa0100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_mcu_r5fss0_core1_dma_memory_region: vision-apps-r5f-dma-memory@a1000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa1000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_mcu_r5fss0_core1_memory_region: vision-apps-r5f-memory@a1100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa1100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core0_dma_memory_region: vision-apps-r5f-dma-memory@a2000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa2000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core0_memory_region: vision-apps-r5f-memory@a2100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa2100000 0x00 0x01f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core1_dma_memory_region: vision-apps-r5f-dma-memory@a4000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa4000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss0_core1_memory_region: vision-apps-r5f-memory@a4100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa4100000 0x00 0x01f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core0_dma_memory_region: vision-apps-r5f-dma-memory@a6000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa6000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core0_memory_region: vision-apps-r5f-memory@a6100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa6100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core1_dma_memory_region: vision-apps-r5f-dma-memory@a7000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa7000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_main_r5fss1_core1_memory_region: vision-apps-r5f-memory@a7100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa7100000 0x00 0x00f00000>;
+		no-map;
+	};
+	vision_apps_rtos_ipc_memory_region: vision-apps-rtos-ipc-memory-region@a8000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xa8000000 0x00 0x02000000>;
+		no-map;
+	};
+	vision_apps_memory_region: vision-apps-dma-memory@aa000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xaa000000 0x00 0x06000000>;
+		no-map;
+	};
+	vision_apps_c71_0_dma_memory_region: vision-apps-c71-dma-memory@b0000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb0000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_c71_0_memory_region: vision-apps-c71_0-memory@b0100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb0100000 0x00 0x05f00000>;
+		no-map;
+	};
+	vision_apps_c71_1_dma_memory_region: vision-apps-c71_1-dma-memory@b6000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb6000000 0x00 0x00100000>;
+		no-map;
+	};
+	vision_apps_c71_1_memory_region: vision-apps-c71_1-memory@b6100000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb6100000 0x00 0x01f00000>;
+		no-map;
+	};
+	vision_apps_core_heaps_lo: vision-apps-core-heap-memory-lo@b8000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00 0xb8000000 0x00 0x04000000>;
+		no-map;
+	};
+	vision_apps_core_heaps_hi: vision-apps-core-heap-memory-hi@880000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x08 0x80000000 0x00 0x2c000000>;
+		no-map;
+	};
+	vision_apps_shared_region: vision_apps_shared-memories {
+		compatible = "dma-heap-carveout";
+		reg = <0x09 0x00000000 0x00 0x20000000>;
+	};
+	linux,cma {
+                status = "disabled";
+        };
+	linux_cma_region: linux-cma-buffers@980000000 {
+		compatible = "shared-dma-pool";
+                reusable;
+                reg = <0x09 0x80000000 0x00 0x38000000>;
+                linux,cma-default;
+	};
+};
+
+&mcu_r5fss0_core0 {
+	memory-region = <&vision_apps_mcu_r5fss0_core0_dma_memory_region>,
+			<&vision_apps_mcu_r5fss0_core0_memory_region>;
+};
+
+&mcu_r5fss0_core1 {
+	memory-region = <&vision_apps_mcu_r5fss0_core1_dma_memory_region>,
+			<&vision_apps_mcu_r5fss0_core1_memory_region>;
+};
+
+&main_r5fss0_core0 {
+	memory-region = <&vision_apps_main_r5fss0_core0_dma_memory_region>,
+			<&vision_apps_main_r5fss0_core0_memory_region>;
+};
+
+&main_r5fss0_core1 {
+	memory-region = <&vision_apps_main_r5fss0_core1_dma_memory_region>,
+			<&vision_apps_main_r5fss0_core1_memory_region>;
+};
+
+&main_r5fss1_core0 {
+	memory-region = <&vision_apps_main_r5fss1_core0_dma_memory_region>,
+			<&vision_apps_main_r5fss1_core0_memory_region>;
+};
+
+&main_r5fss1_core1 {
+	memory-region = <&vision_apps_main_r5fss1_core1_dma_memory_region>,
+			<&vision_apps_main_r5fss1_core1_memory_region>;
+};
+
+&c71_0 {
+	memory-region = <&vision_apps_c71_0_dma_memory_region>,
+			<&vision_apps_c71_0_memory_region>;
+};
+
+&c71_1 {
+	memory-region = <&vision_apps_c71_1_dma_memory_region>,
+			<&vision_apps_c71_1_memory_region>;
+};
diff --git a/arch/arm64/boot/dts/ti/k3-j721s2-vision-apps.dtso b/arch/arm64/boot/dts/ti/k3-j721s2-vision-apps.dtso
new file mode 100644
index 000000000000..6d6179ff6510
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-j721s2-vision-apps.dtso
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * vision-apps: device-tree overlay
+ *
+ * Copyright (C) 2019 - 2021 Texas Instruments Incorporated - http://www.ti.com/
+ */
+
+/dts-v1/;
+/plugin/;
+
+#include "k3-j721s2-rtos-memory-map.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+
+&main_i2c1 {
+    status = "disabled";
+};
+
+&main_i2c4 {
+	status = "disabled";
+};
+
+&main_i2c5 {
+	status = "disabled";
+};
+
+&serdes_wiz0 {
+    status = "disabled";
+};
+
+&dss {
+    status = "disabled";
+};
+
+&ti_csi2rx0 {
+    status = "disabled";
+};
+
+&ti_csi2rx1 {
+    status = "disabled";
+};
+
+&exp_som {
+	P5_hog {
+		gpio-hog;
+		gpios = <5 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "GPIO_eDP_ENABLE";
+	};
+};
-- 
2.34.1

